wpimath/src/main/native/include/frc/geometry/Rotation3d.h

```diff
@@ -63,15 +99,72 @@ class WPILIB_DLLEXPORT Rotation3d {
    *
    * @param rvec The rotation vector.
    */
-  explicit Rotation3d(const Eigen::Vector3d& rvec);
+  constexpr explicit Rotation3d(const Eigen::Vector3d& rvec)
+      : Rotation3d{rvec, units::radian_t{ct_matrix{rvec}.norm()}} {}
 
   /**
    * Constructs a Rotation3d from a rotation matrix.
    *
    * @param rotationMatrix The rotation matrix.
    * @throws std::domain_error if the rotation matrix isn't special orthogonal.
    */
-  explicit Rotation3d(const Eigen::Matrix3d& rotationMatrix);
+  constexpr explicit Rotation3d(const Eigen::Matrix3d& rotationMatrix) {
+    // Require that the rotation matrix is special orthogonal. This is true if
+    // the matrix is orthogonal (RRᵀ = I) and normalized (determinant is 1).
+    auto impl = []<typename Matrix3d>(const Matrix3d& R) -> Quaternion {
+      if ((R * R.transpose() - Matrix3d::Identity()).norm() > 1e-9) {
```

Comment:
```suggestion
    auto impl = []<typename Matrix3d>(const Matrix3d& R) -> Quaternion {
      // Require that the rotation matrix is special orthogonal. This is true if
      // the matrix is orthogonal (RRᵀ = I) and normalized (determinant is 1).
      if ((R * R.transpose() - Matrix3d::Identity()).norm() > 1e-9) {
```

Reply:
I want to avoid runtime overhead whenever possible.